community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/RelationshipCountsProcessor.java

```diff
@@ -55,28 +58,17 @@ public RelationshipCountsProcessor( NodeLabelsCache nodeLabelCache,
         // Make room for high id + 1 since we need that extra slot for the ANY counts
         this.anyLabel = highLabelId;
         this.anyRelationshipType = highRelationshipTypeId;
-        this.itemsPerType = anyLabel+1;
-        this.itemsPerStartLabel = (anyRelationshipType+1)*itemsPerType;
-        this.counts = cacheFactory.newLongArray( arrayIndex( highLabelId, highRelationshipTypeId, highLabelId )+1, 0 );
-    }
-
-    private long arrayIndex( long startLabel, long relationshipType, long endLabel )
-    {
-        return startLabel*itemsPerStartLabel + relationshipType*itemsPerType + endLabel;
-    }
-
-    private void increment( long startLabel, long relationshipType, long endLabel )
-    {
-        long index = arrayIndex( startLabel, relationshipType, endLabel );
-        counts.set( index, counts.get( index ) + 1 );
+        this.itemsPerType = anyLabel + 1;
+        this.itemsPerLabel = anyRelationshipType + 1;
+        this.labelsCounts = cacheFactory.newLongArray( sideSize() * SIDES, 0 );
+        this.wildcardCounts = cacheFactory.newLongArray( anyRelationshipType + 1, 0 );
     }
```

Comment:
Stripping back the indirections in the size calculations on both sides, and treating it just as formulae based on the input parameters, we get the following:

The old `counts` had a size of:

    (highLabelId + 1) * (highLabelId + 1) * (highRelationshipTypeId + 1)

The new sizing is divided over two arrays, `labelsCounts` and `wildcardCounts`, where the sizing is as follows:

For `labelsCounts` the size is:

    2 * (highLabelId + 1) * (highRelationshipTypeId + 1)

For `wildcardCounts` the size is:

    highRelationshipTypeId + 1

That means the total size in the new code is:

    (2 * highLabelId + 3) * (highRelationshipTypeId + 1)

So this is means a good improvement in size of the underlying arrays.